packages/editor-sdk/src/components/Widgets/BooleanField.tsx

```diff
@@ -12,14 +12,15 @@ declare module '../../types/widget' {
 }
 
 export const BooleanField: React.FC<WidgetProps<BooleanFieldType>> = props => {
-  const { value, onChange } = props;
+  const { value, onChange, path } = props;
 
   useEffect(() => {
     // Convert value to boolean after switch from expression widget mode.
-    if (typeof value !== 'boolean') {
-      onChange(true);
+    if (value && typeof value !== 'boolean') {
+      const convertValue = path[0] === 'ifCondition';
```

Comment:
I think we don't need treat `ifCondition` specially.

Reply:
Yes, here it should only be necessary to avoid accidental conversions     
changed